compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -859,6 +859,15 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
             && let hir::Node::Expr(lhs) = self.tcx.hir_node(*lhs_hir_id)
             && let hir::Node::Expr(rhs) = self.tcx.hir_node(*rhs_hir_id)
             && let Some(rhs_ty) = typeck_results.expr_ty_opt(rhs)
+            && let trait_pred = predicate.unwrap_or(trait_pred)
+            // Only run this code on binary operators
+            && hir::lang_items::OPERATORS
+                .iter()
+                .filter(|op| !matches!(op, LangItem::Neg | LangItem::Not))
```

Comment:
I really dislike hard-coding the unary operators here. Please make a new group of `lang_items::BINARY_OPERATORS` or something instead.

Reply:
Done, thanks.